src/Microsoft.AspNetCore.JsonPatch/Converters/JsonPatchDocumentConverter.cs

```diff
@@ -72,5 +72,14 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
                 serializer.Serialize(writer, lst);
             }
         }
+
+        /// <summary>
+        /// Create the JsonPatchDocument using the appropriate constructor
+        /// </summary>
+        /// <param name="operations">The operations to assign to the JsonPatchDocument</param>
+        /// <returns>A new instance of the JsonPatchDocument</returns>
+        protected virtual JsonPatchDocument CreateContainer(List<Operation> operations ) =>
```

Comment:
I'm not totally sure what this is here for - how do you expect this to be used?

Reply:
@rynowak We added this purely for extension purposes.  If I remember correctly @HappyNomad needed a custom JsonPatchDocumentConverter, and breaking out this portion to be overridden allowed the rest of the ReadJson function to be used rather than being duplicated.

Reply:
Right, what are the scenarios for a custom converter? What problems are being solved?

Reply:
Looking back through the thread, @HappyNomad needed to provide his custom adapter factory as part of the deserialization process.  I didn't need to for my scenario, and to be honest I never went back to understand why one of us needed it and the other didn't.

Reply:
@HappyNomad - care to reply?

I don't think this is blocking if we don't get a response. I just like to know what's going on 😆

Reply:
> Right, what are the scenarios for a custom converter? What problems are being solved?

For an answer, see https://github.com/aspnet/JsonPatch/pull/132#issuecomment-372409378 .

But as long as I can pass `IAdapterFactory` to `ApplyTo`, I won't need it.  See https://github.com/aspnet/JsonPatch/pull/132#issuecomment-373957371 .